### PR TITLE
Better tab completes

### PR DESCRIPTION
**Pull request by stzups**
#### Description: 

See commits for individual changes.
This PR changes commands to:
- Be more readable
- Return more relevant subcommands/residents/towns/nations/worlds
- Add missing features for certain subcommands, like /t invite
- Return empty list instead of null (online players) when no completions match

Fixed Trie to not be case sensitive when searching, and to respect case when adding new keys. Also fixed adding only lowercase strings in TownyUniverse.
____
- [x] I have tested this pull request for defects on a server. 

By making this pull request, I represent that I have the right to waive copyright and related rights to my contribution, and agree that all copyright and related rights in my contributions are waived, and I acknowledge that the TownyAdvanced organization has the copyright to use and modify my contribution under the Towny [License](https://github.com/LlmDl/Towny/blob/master/LICENSE.md) for perpetuity.

### CODE REVIEW

**Comment by stzups** [comment]
found some issues with the trie, fixing now

**Comment by stzups** [comment]
Fixed. Upper/lowercase issues are resolved, /t join was added, and the Trie is now built once instead of twice. The keys for the trie are added when residents, nations, and towns are loaded from the db, so there is no need to build the trie again at the end of the load function in TownyUniverse

File: src/com/palmergames/bukkit/towny/command/InviteCommand.java [hunk 1]
```diff
@@ -55,22 +55,21 @@ public InviteCommand(Towny instance) {
...
-							.collect(Collectors.toList()), args[1]);
-					} catch (TownyException ignored) {}
-			}
+		switch (args.length) {
+			case 1:
+				return NameUtil.filterByStart(inviteTabCompletes, args[0]);
+			case 2:
+				switch (args[0].toLowerCase()) {
+					case "accept":
+					case "deny":
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/command/InviteCommand.java]: One thing I forgot to mention last time around was that accept/deny are configurable in the config. I dunno if we want to handle that.

File: src/com/palmergames/bukkit/towny/command/TownyCommand.java [hunk 2]
```diff
@@ -57,11 +58,22 @@
...
 		"top",
 		"spy",
-		"tree",
 		"universe",
 		"v",
 		"war"
 	));
+	
+	private static final List<String> townyConsoleTabCompletes = new ArrayList<>(Arrays.asList(
+		"map",
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/command/TownyCommand.java]: map is one that cannot be used from console

File: src/com/palmergames/bukkit/towny/command/InviteCommand.java (hunk 1 above)
Comment (stzups) [reply src/com/palmergames/bukkit/towny/command/InviteCommand.java]: We do

**Pull request merged**